archive/dug_6/dug_6_dasharo_configuration_utility_status_update.md

```diff
@@ -0,0 +1,360 @@
+- ~~Chipsec and HSI Checks Integration~~
+- ~~Integration with Commercial Tools like Binarly~~
+
+<br>
+
+.center.image-35[![](/img/slacking_off.png)]
+
+???
+
+- And here is what we actually delivered.
+- But that doesn't mean we've been inactive.
+
+---
+
+# The problem
+
+.center[
+<img src="/img/x86_ring_level_arrow.svg" height="550px" style="margin-top:-70px; margin-left:20px">
+]
+
+???
+
+- _SMM is a special operating mode in x86 processors that handles system-wide
+functions like:_
+  - _power management_
+  - _system hardware control_
+  - _firmware settings storage_
+- To be precise settings are stored on SMM mediated part of SPI memory.
+- This separates this funtionality from the userspace protecting it from
+  software attacks
+- coreboot stores it's firmware settings in the SMM
+  - Until recently they could be only modified by hand from the UEFI menu
+  - So we made a tool for changing the settings via an util/smmstoretool
+- We had to develop a way to access the contents of the dasharo binary without
+  manually setting the UEFI variables.
+- Fun fact: so new the documentation still wasn't merged
+
+---
+
+# Our solution
+
+
+.center[
+### [smmstoretool](https://github.com/coreboot/coreboot/tree/main/util/smmstoretool)
+<img src="/img/coreboot_repo_smmstoretool_sc.png" height="390px" style="margin-left:100px">
+]
+
+???
+
+- Here at 3mdeb we've developed and upstreamed the `smmstoretool` utility for
+  offline modification of the SMM storage in the coreboot or Dasharo binaries
+- In other words we can now set our firmware settings without booting it up
+
+---
+
+# What does this mean to Dasharo users
+
+<!-- Accepted values are slanted so they appear correctly on the presentation -->
+
+.code-15px[
+- **Get a list of settings in a binary**:
+```sh
+$ ./dcuc variable --list coreboot.rom
+Settings in coreboot.rom:
+NAME		VALUE		    ACCEPTED VALUES
+MeMode	  Enabled          Enabled / Disabled (Soft) / Disabled (HAP)
+```
+- **Change a setting**:
+```sh
+$ ./dcuc variable coreboot.rom --set "MeMode" --value "Disabled (Soft)"
+```
+```sh
+$ ./dcuc variable --list coreboot.rom
+Settings in coreboot.rom:
+NAME		VALUE			ACCEPTED VALUES
+MeMode	  Disabled (Soft)  Enabled / Disabled (Soft) / Disabled (HAP)
+```
+]
+
+???
+
+- We expanded the DCU with additional utilities based on the smmstoretool
+- This allows for easy modification of binaries and the firmware settings
+
+#### Use Cases
+
+- Turning on serial support from the start for headless platforms
+or for automatic testing
+- Set power state after failure so the board boots automatically when the power
+to supply unit is delivered - also useful for some test cases.
+- Set the network boot to enabeld (defaults to disabled). So after flashing we
+can readily boot from network (useful in manufacturing, testing,
+OS installation).
+- Change any setting you want from the OS rather than from firmware menu.
+  -It can be done as long as the SMM Bios Write Protection is not locked.
```

Comment:
does it render correctly?

Reply:
[Fixed](https://github.com/Dasharo/presentations/blob/021712068a8e4eb68c8546d90d2cf8e3ba8ddda2/archive/dug_6/dug_6_dasharo_configuration_utility_status_update.md?plain=1#L278)